docs/testplan/HA-SmartSwitch-test-plan.md

```diff
@@ -0,0 +1,262 @@
+            "PRIORITY": "1",
+            "SRC_IP": <DPU1 IP>,
+            "DST_IP": <NPU1 IP>,
+            "IP_TYPE": "IP",CO
+            "L4_SRC_PORT": "3784"
+        }
+     }
+```
+
+
+### Module 6 Critical Process Crash 
+For all process crash cases, we will have 4 variations, it’s 
+1. Process crash on DPU1, traffic landing on DPU2
+2. Process crash on DPU2, traffic landing on DPU1
+3. Process crash on DPU1, traffic landing on DPU1
+4. Process crash on DPU2, traffic landing on DPU2   
+
+The expected behavior is same, that HA state remains unchanged.
+
+| Case          | Goal                                     | Test Steps                                      | Expected Control Plane Behavior            | Expected Data Plane Behavior                                                                                                                                                               |
+| ------------- | ---------------------------------------- | ----------------------------------------------- | ------------------------------------------ | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| syncd on DPU  | Verify when syncd crash on DPU.          | • Start sending traffic<br>• Kill syncd on DPU  | DPU1 remains active, DPU2 remains standby. | T2 receives packets with allowed disruption.                                                                                                                                               |
+| hamgrd on NPU | Verify when hamgrd crash on NPU.         | • Start sending traffic<br>• Kill hamgrd on NPU | DPU1 remains active, DPU2 remains standby. | T2 receives packets with allowed disruption.                                                                                                                                               |
+| pmon on NPU   | Verify when pmon crash on NPU.           | • Start sending traffic<br>• Kill pmon on NPU   | DPU1 remains active, DPU2 remains standby. | T2 receives packets with allowed disruption.                                                                                                                                               |
+| bgpd on NPU   | Verify when bgpd crash on NPU.           | • Start sending traffic<br>• Kill bgpd on NPU   | DPU1 remains active, DPU2 remains standby. | T2 receives packets with allowed disruption.                                                                                                                                               |
+| bgpd on DPU   | Verify when frr is killed on active DPU. | • Start sending traffic<br>• Kill bgpd on DPU   | DPU1 remains active, DPU2 remains standby. | T2 receives packets with allowed disruption. (BFD runs in FRR on DPU, hence BFD state will be down for local DPU probing, traffic will be sniffed from standby side after bgpd is killed.) |
```

Comment:
frr container kill on DPU is also tricky especially in the DPU-driven mode since the BFD and the HA state can go out of sync about who is the Active DPU. Is DPU reboot a simpler option for these critical process crash scenarios?

Reply:
Same as above. Updated the expected control plane behavior.